### PR TITLE
feat: add governance

### LINKED ISSUE #19: Create a shared `GOVERNANCE.md`
Write down governance rules for this community. Cover following topics:

*This is a brain dump, feel free to extend.*

- [x] How to become a member:
   - Create and commit to maintaining a plugin - becomes code owner of that plugin
   - Asks for it and it's approved by project area maintainer to join a wider project area group
   - Is nominated by a maintainer
- [x] Who is a maintainer
- [x] Define project areas
- [x] Contributor roles are managed through github groups
- [x] Repositories have github groups serving as code owners per area
- [x] Access is reviewed annually
- [x] Member who is not active will lose merge/review permissions and will be recognized as a honorable/emerit member of the team (remove from group but left in org and added to a running doc of recognized members/authors)

### PR DESCRIPTION
**Pull request by tumido**
Resolves: #19 

cc @jasperchui @serenamarie125 @gorkem @christophe-f  @rtaniwa 

the idea is to create the following GitHub teams. Their responsibilities are defined in `GOVERNANCE.md`

- `@janus-idp/steering` with admin access to all. Members: Serena, Gorkem, Christophe, Rob, Tom
- `@janus-idp/maintainers` parent team with following children:
   - `@janus-idp/maintainers-plugins` - write access to `backstage-plugins` (individual `CODEOWNERS` per plugin may exist)
   - `@janus-idp/maintainers-showcase` - write access to `backstage-showcase`
   - `@janus-idp/maintainers-helm` - write access to `helm-backstage`
   - `@janus-idp/maintainers-gpt` - write access to `software-templates`
   - `@janus-idp/maintainers-demo` - write access to `demo-setup`
   - `@janus-idp/maintainers-website` - write access to `janus-idp.github.io`
   - `@janus-idp/maintainers-blog` - `CODEOWNERS` for `.md[x]` in `janus-idp.github.io` repo, has write access to this repository.
   - `@janus-idp/maintainers-docs` - `CODEOWNERS` for `.md` files
- `@janus-idp/issue-triage` for people with triage access

### CODE REVIEW

**Comment by SecKatie** [comment]
This is great! However, to fully comply with the standard, we must include one additional detail around reviewing access on a regular cadence. It must be done at least once a year.

Something like this could work:

```markdown
## Annual Access Review

The TSC will review access to repositories at least annually. The purpose of this review is to ensure that only authorized individuals have access to the repositories and that they have the appropriate permissions.

During the review, the TSC will:

1. Verify the list of people with access to the repositories and their respective permissions.
2. Check for any unauthorized access, such as team members who have left the project but still have access to the repositories.
3. Document the date of the review, the list of repositories reviewed, and the results of the review.

If any unauthorized access is detected, the TSC will take corrective actions and notify the Red Hat Information Security team for assistance.

The TSC will maintain records of review results, including the date the review was performed and any changes made to repository access.
```

**Comment by rtaniwa** [comment]
FWIW i also did a review today of our current members, and there was some clean up done. So I think once we merge this, we should be good to go.

**Review by christophe-f — APPROVED** [review]
LGTM

**Comment by tumido** [comment]
@JoshuaMulliken I've applied your suggestion, thank you for this! However, I did a few changes 🙂 

```diff
- 2. Check for any unauthorized access, such as team members who have left the project but still have access to the repositories.
+ 2. Check for any unauthorized access, such as team members who have left the project but still have WRITE access to the repositories.
```

I think we should make this more specific. It's not a good habit for a community to completely remove members when they leave the project, we should just take preventive measures so they are not allowed to force changes by themselves.

```diff
- If any unauthorized access is detected, the TSC will take corrective actions and notify the Red Hat Information Security team for assistance.
+ If any unauthorized access is detected, the TSC will take corrective actions and consult the Red Hat Information Security team that offered their assistance and support.
```

I think we should word this in a way that we're not forcing on the community that it somehow reports to the Red Hat Information Security team. Instead, we should say something like that it's a voluntary consultation initiated by the TSC of their free will, and the infosec team is here not he make demands but rather to provide support. I'm not saying your sentence is worded like that, but I can see it being misunderstood that way. WDYT?

**Comment by tumido** [comment]
@serenamarie125 @rtaniwa @christophe-f can you please review/approve?

**Comment by christophe-f** [comment]
> @serenamarie125 @rtaniwa @christophe-f can you please review/approve?

Already approved

**Pull request merged**
